include/random_walks/compute_diameter.hpp

```diff
@@ -40,6 +37,16 @@ struct compute_diameter<HPolytope<Point, MT>>
     }
 };
 
+template <typename Point>
+struct compute_diameter<const HPolytope<Point>>
+{
+    template <typename NT>
+    static NT compute(HPolytope<Point> const& P)
+    {
+        return NT(2) * std::sqrt(NT(P.dimension())) * P.InnerBall().second;
```

Comment:
I think we should call ComputeInnerBall() from the Polytope.
We have a flag in Hpolytope that checks if the innerball is already computed in ComputeInnerBall().
If the ball isn't computed then InnerBall() will trigger a runntime error.
I would probably add a todo to do the same check in InnerBall() as well and if the ball is not computed then we call ComputeInnerBall() in InnerBall(). Or we could delete InnerBall() and rename the ComputeInnerBall() to InnerBall().